parser/parser.go

```diff
@@ -500,6 +507,36 @@ func (p *parser) parseStringValue() *String {
 	return value
 }
 
+func (p *parser) parseIntValue() *Int64 {
+	var str string
+	literalPos := p.scanner.Position
+	if p.tok == '-' {
+		if p.scanner.Peek() < 48 || p.scanner.Peek() > 57 {
```

Comment:
int('0') and int('9')

why not just p.accept(p.tok) and then check that the new p.tok is a scanner.Int?

Reply:
I think p.accept() allows scanner.comment and whitespace. since I don't hope any comments inserted in between "-" and int, so that's why I use scanner.Peek()

Reply:
We accept comments almost anywhere between tokens, I don't see why this should be different.  You can enforce putting them after the in the printer package, which will fix them in bpfmt.

Reply:
ok. removed this check to just use p.accept().